Integrate scan_fmt_scxml crate with SCXML parsing logic

pouyapd opened the pull request:
This pull request integrates a new crate, `scan_fmt_scxml`, to add SCXML parsing capabilities to the SCAN project.

Changes include:
- Added the `scan_fmt_scxml` crate containing the SCXML parsing logic.
- Modified `scan_scxml/src/parser.rs` to use the `scan_fmt_scxml` library for parsing .scxml files.
- Updated root `Cargo.toml` to include `scan_fmt_scxml` in the workspace and dependencies.
- Updated `scan_scxml/Cargo.toml` to add `scan_fmt_scxml` as a dependency for the `scan_scxml` crate.

This allows SCAN to load models specified in the SCXML format.

EnricoGhiorzi commented:
Dear @pouyapd thank you for the PR. There are a few issues with it:

- The commit has no "sign-off" which is why is failing the DCO test. You need to sign-off all of your commits to attest you are submitting your code according to the repo licence. Please look at online documentation about this git sign-off functionality and how to fix your current commit.
- The code does not build because `scan_fmt_scxml` has no `Cargo.toml` manifest. I haven't tried building past this point, so I don't know if the rest works.
- You are importing `scan_fmt_scxml` into `scan_scxml` as a dependency, which should not be required. On the contrary, you might want to import `scan_scxml` into `scan_fmt_scxml` to use the builder functionalities.
- The PR is composed by a single massive commit (by the way, that's not best practice, but I am fine with it). The real issue is that this way your colleagues do not appear as contributors. Please ensure that all of you push some commits (with sign-off) so that you will appear as contributors in the repo.

Please fix the above issues so that I can move on with the review process.

pouyapd commented:
Dear Professor,

I have reviewed your feedback on this Pull Request.

I have made the following corrections:
- Fixed the DCO sign-off issue. The latest commit is now correctly signed.
- Added the missing Cargo.toml manifest file for the `scan_fmt_scxml` crate.
- Corrected the dependency in `scan_scxml/Cargo.toml.
- The automated workflow is now running (or waiting for approval) and should pass the basic build check related to Cargo.toml files and sign-off.

Thank you again for your clear feedback!

Best regards,

EnricoGhiorzi commented on scan_fmt_scxml/Cargo.toml:
```diff
@@ -0,0 +1,24 @@
+[package]
+name = "scan_scxml"
```
The name conflicts with the other `scan_scxml`. I suggest renaming your package and library crate as `scan_fmt_scxml`.

EnricoGhiorzi commented on scan_fmt_scxml/Cargo.toml:
```diff
@@ -0,0 +1,24 @@
+edition = "2024"
+
+[lib]
+name = "scan_scxml"  # The name of the target.
```
Same as previous comment.

EnricoGhiorzi commented on scan_fmt_scxml/Cargo.toml:
```diff
@@ -0,0 +1,24 @@
+chumsky = "0.10.1"
+csv = { workspace = true }
+flate2 = { workspace = true }
+scan_fmt_scxml = { path = "../scan_fmt_scxml" }
```
A package cannot have itself as a dependency. You maybe wanted to import `scan_scxml`? Not sure it's needed anyway.

EnricoGhiorzi reviewed (changes requested):
There are issues with the `scan_fmt_scxml/Cargo.toml` file that prevent building. Please fix these so that I can move on with the review.

pouyapd commented:
Dear Professor,
Following up on Pull Request https://github.com/convince-project/scan/pull/36. The DCO check has passed.
The automated Rust build workflow is currently still pending approval on GitHub before it can run. Please let me know if you are able to approve it so I can see the build results.

EnricoGhiorzi reviewed (changes requested):
Dear SCXML group, I have reviewed the PR and I found some issues:

- The code does not compile: there are issues on the `Cargo.toml` manifests breaking the compilation. Did you  forgot to include some commits maybe?
-  I made the code compile and run the tests with `cargo test`. I found out that the tests in `scxml_lib` do not run because there is no associated `test.rs` file. Please include such a file (you can take as example the relevant code in the other packages) so that your code can be automatically tested.
